flexget/plugins/plugin_sftp.py

```diff
@@ -0,0 +1,423 @@
+
+# retry configuration contants
+CONNECT_TRIES = 3
+RETRY_INTERVAL = 15
+RETRY_STEP = 5
+
+try:
+    import pysftp
+    logging.getLogger("paramiko").setLevel(logging.ERROR)
+except:
+    pysftp = None
+
+
+def sftp_connect(conf):
+    """
+    Helper function to connect to an sftp server
+    """
+    sftp = None
+    tries = CONNECT_TRIES
+    retry_interval = RETRY_INTERVAL
+
+    while not sftp:
+        try:
+            sftp = pysftp.Connection(host=conf.host, username=conf.username,
+                                     private_key=conf.private_key, password=conf.password, 
+                                     port=conf.port, private_key_pass=conf.private_key_pass)
+            log.debug('Connected to %s' % conf.host)
+        except Exception as e:
+            if not tries:
+                raise e
+            else:
+                log.debug('Caught exception: %s' % e)
+                log.warn('Failed to connect to %s; waiting %d seconds before retrying.' % 
+                         (conf.host, retry_interval))
+                time.sleep(retry_interval)
+                tries -= 1
+                retry_interval += RETRY_STEP
+    
+    return sftp
+
+
+def dependency_check():
+    """
+    Check if pysftp module is present
+    """
+    if not pysftp:
+        raise plugin.DependencyError(issued_by='sftp', 
+                                     missing='pysftp', 
+                                     message='sftp plugin requires the pysftp Python module.')
+
+
+class SftpList(object):
+    """
+    Generate entries from SFTP. This plugin requires the pysftp Python module and its dependencies.
+
+    Configuration:
+
+    config
+        host:                 Host to connect to
+        port:                 Port the remote SSH server is listening on. Defaults to port 22.
+        username:             Username to log in as
+        password:             The password to use. Optional if a private key is provided.
+        private_key:          Path to the private key (if any) to log into the SSH server
+        private_key_pass:     Password for the private key (if needed)
+        recursive:            Indicates whether the listing should be recursive
+        get_size:             Indicates whetern to calculate the size of the remote file/directory.
+                              WARNING: This can be very slow when computing the size of directories!
+        files_only:           Indicates wheter to omit diredtories from the results.
+    dirs:
+        List of directories to download
+
+    Example:
+
+      sftp_list:
+          config:
+              host: example.com
+              username: ieaston
+              private_key: /Users/uername/.ssh/id_rsa
+              recursive: False
+              get_size: True
+              files_only: False
+          dirs:
+              - '/path/to/list/'
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'config': {
```

Comment:
I just did it that way to keep it similar to the existing FTP/FTPS plugin. Not sure why the author of that plugin did it that way, so I'm not particularly attached to the idea.